### PR TITLE
feat: make priority scheduler configurable and add docs [DET-4641]

### PR DESCRIPTION
**Pull request by aaron276h**
This PR should not be landed until after #1639 lands and dogfooding is completed.

### CODE REVIEW

File: docs/reference/cluster-config.txt [hunk 1]
```diff
@@ -230,8 +230,27 @@ The master supports the following configuration settings:
          available resources depending on the resource they require and
          their weight.
 
-      -  ``priority``: Tasks are scheduled in the order of the order in
-         which they arrive at the cluster.
+      -  ``round_robin``: Tasks are scheduled in the order of the order
+         in which they arrive at the cluster.
+
+      -  ``priority``: Tasks are scheduled based on their prirority.
```
[the remark this answers is not in the document]
Comment (eecsliu) [reply docs/reference/cluster-config.txt:236]: suggestion: Tasks are scheduled based on their priority, which can range from the values 1 to 99 inclusive.

File: master/internal/resourcemanagers/scheduler.go [hunk 2]
```diff
@@ -13,15 +13,15 @@ type Scheduler interface {
...
-	switch schedulingPolicy {
+func MakeScheduler(config *SchedulerConfig) Scheduler {
+	switch config.GetType() {
 	case priorityScheduling:
-		return NewRoundRobinScheduler()
+		return NewPriorityScheduler(config)
 	case fairShareScheduling:
 		return NewFairShareScheduler()
 	case roundRobinScheduling:
 		return NewRoundRobinScheduler()
```
Comment (eecsliu) [inline master/internal/resourcemanagers/scheduler.go:23]: Are we going to make another case for round robin scheduler?

File: docs/reference/cluster-config.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (aaron276h) [reply docs/reference/cluster-config.txt]: good call, updated this

File: docs/reference/cluster-config.txt [hunk 3]
```diff
@@ -230,8 +230,27 @@ The master supports the following configuration settings:
...
+         Tasks can be assigned priorities from 1 to 99 inclusively.
+         Lower priority numbers indicate higher priority tasks. A lower
+         priority task will never be scheduled while a higher priority
+         task is pending. Zero-slot tasks (e.g., cpu-only notebooks,
+         tensorboards) are prioritized separately from tasks requiring
+         slots (e.g., experiments running on GPUs). Task priority can be
+         assined using the the ``resources.priority`` field. If a task
```
[the remark this answers is not in the document]
Comment (aaron276h) [reply docs/reference/cluster-config.txt]: good catch

File: docs/reference/cluster-config.txt [hunk 4]
```diff
@@ -230,8 +230,27 @@ The master supports the following configuration settings:
...
+         does not specify a priority it is assigned the
+         ``default_priority``.
+
+         -  ``preemption``: Specifies whether lower priority tasks
+            should be preempted to schedule higher priority tasks. Tasks
+            of the lowest priority are preempted first.
```
[the remark this answers is not in the document]
Comment (aaron276h) [reply docs/reference/cluster-config.txt]: done

File: master/internal/resourcemanagers/scheduler.go (hunk 2 above)
Comment (aaron276h) [reply master/internal/resourcemanagers/scheduler.go:23]: Why do we need another case?

Comment (eecsliu) [reply master/internal/resourcemanagers/scheduler.go:23]: I need to see an optometrist - we're good, for some reason I only saw two cases, not the round robin one.

Comment (aaron276h) [reply master/internal/resourcemanagers/scheduler.go:23]: LMAO

**Review by eecsliu — APPROVED** [review]
👌

**Pull request merged**